client/order/index.js

```diff
@@ -175,16 +175,20 @@ jQuery( function ( $ ) {
 			'#wcpay-order-payment-details-container'
 		);
 
-		// If the container doesn't exist (WC < 7.9), or the charge ID isn't present, don't render the notice.
+		// Check if the order's test mode meta matches the site's current test mode setting.
+		// E.g. order and site are both in test mode, or both in live mode.
+		const isOrderTestModeMatch = orderTestMode === wcpaySettings.testMode;
```

Comment:
In order to continue to display the disputed order notice for orders created < 6.9, we could check if the `_wcpay_mode` order meta exists before checking for a match. This would mean that the network error will continue to occur for orders created < 6.9 if the order/site test mode is mismatched.